cadence/contracts/utility/NFTForwarding.cdc

```diff
@@ -0,0 +1,88 @@
+/// ## Non-Fungible Token Forwarding Contract
+///
+/// This contract enables a user to designate a recipient to NFTs could be forwarded
+///
+/// The NFTForwarder resource can be referenced just like any NonFungibleToken Receiver,
+/// allowing a sender to deposit NFTs as they usually would
+///
+/// However, in this implementation, any time a deposit is made, the deposited NFT is
+/// additionally deposited to a predefined recipient.
+///
+/// To create an NFTForwarder resource, an account calls the createNewNFTForwarder
+/// function, passing the Receiver Capability to which NFTs will be forwarded.
+
+import NonFungibleToken from "../NonFungibleToken.cdc"
+
+pub contract NFTForwarding {
+
+    pub event ForwardedNFTDeposit(id: UInt64, from: Address?)
+    pub event NFTForwarderRecipientChanged(forwarder: Address?)
+
+    /// Canonical Storage and Public paths
+    ///
+    pub let NFTForwarderStoragePath: StoragePath
+
+    /// Resource that forwards deposited NFTs to a designated
+    /// recipient's collection
+    ///
+    pub resource NFTForwarder: NonFungibleToken.Receiver {
+
+        /// Recipient to which NFTs will be forwarded
+        ///
+        access(self) var recipient: Capability
```

Comment:
Better to have defined type here - `Capability<&{NonFungibleToken.CollectionPublic}>`